services/thermal_mgr/thermal_mgr.c

```diff
@@ -41,20 +41,57 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 }
 
+#define MAX_MS_WAIT 20
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
+  xQueueSend(thermalMgrQueueHandle, (void *)event, portTICK_PERIOD_MS*MAX_MS_WAIT);
+  return ERR_CODE_SUCCESS;
+}
 
+error_code_t thermalMgrSendEventISR(thermal_mgr_event_t *event) {
+  /* Send an event to the thermal manager queue */
+  xQueueSendFromISR(thermalMgrQueueHandle, (void *)event, NULL);
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  // Use new thermalMgr event for non-blocking.
+  thermal_mgr_event_t evt = {0};
+  evt.type = THERMAL_MGR_EVENT_OVER_TEMP_CMD;
+  thermalMgrSendEventISR(&evt);
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+  lm75bd_config_t config = *(lm75bd_config_t *) pvParameters;
+  float temp = {0};
+  error_code_t err_code = {0};
+
   while (1) {
-    
+    thermal_mgr_event_t temp_evt = {0};
+    xQueueReceive(thermalMgrQueueHandle, (void *)&temp_evt, portTICK_PERIOD_MS*MAX_MS_WAIT);
```

Comment:
you need to check if this queue receive succeeded or not before doing the switch statement

Reply:
Yep, thanks!